projects/js-packages/components/components/text/index.jsx

```diff
@@ -32,11 +32,41 @@ export const TIPOGRAPHY_SIZES = {
 	small: 'small',
 };
 
+const BOX_MODEL_VALUES = [
+	'mt',
+	'mr',
+	'mb',
+	'ml',
+	'mx',
+	'my',
+	'm',
+	'pt',
+	'pr',
+	'pb',
+	'pl',
+	'px',
+	'py',
+	'p',
+];
+export const SPACING_VALUES = [ 0, 1, 2, 3, 4, 5, 6, 7, 8 ];
+
 const Text = ( { variant, children, component, className, ...componentProps } ) => {
 	const Component = component || VARIANTS_MAPPING[ variant ] || 'span';
-	const componentClassName = classNames( styles[ variant ], className );
+
+	// Build Styles module CSS classnames.
+	const boxModelClasses = BOX_MODEL_VALUES.reduce( ( acc, value ) => {
+		if ( typeof componentProps[ value ] !== 'undefined' ) {
+			acc += styles[ `${ value }-${ componentProps[ value ] }` ] + ' ';
```

Comment:
We need to get directly these props and not use them from `componentProps`, which is used for composition and normally applies directly to first-class tags `<h1>, <p>`, etc ... 

This way we'll probably receive some warnings

Reply:
sorry I don't get it. `componentProps` is _the rest of the component props_, right? In the end, it pulls the props from the function argument.

Reply:
@retrofox The rest props are designed to pass down props when you're using the `component` prop. The `component` props are designed to be used for the cases that you want the `text` styles and not use its default `component` for that moment.

Example of this usage:

https://github.com/Automattic/jetpack/blob/master/projects/packages/my-jetpack/_inc/components/product-detail-card/index.jsx#L218-L226

In this case, we're using the `Text` component only to get the `typography` styles, but the final component that renders is `ProductDetailButton`, and we pass some props that are for `ProductDetailButton` and not `Text`.

That's the reason we get what we need at `Text` and pass down only the rest.

The other reason is that `Unknown Prop Warning`: https://reactjs.org/warnings/unknown-prop.html 

Is a warning related to passing non-standard props for `DOM` elements, and we're mostly based on `DOM` elements.

Reply:
I think I get that. The idea is picking those props that in some way belong to the `<text />` component. It means typography but also now spacing props. The rest of props should be pass down to the component:

```jsx
<Text
	component={ ProductDetailButton }
	onClick={ clickHandler }
	isLoading={ isFetching }
	isPrimary={ ! isBundle }
	href={ onClick ? undefined : addProductUrl }
	className={ styles[ 'checkout-button' ] }
	variant="body"
>
	Click me!
</Text>
``` 

Clearly, above, the props that belong to the Text component are `variant` and `component`, correct? We could do something like...

```jsx
<Text
	component={ ProductDetailButton }
	onClick={ clickHandler }
	href={ onClick ? undefined : addProductUrl }
	className={ styles[ 'checkout-button' ] }
	variant="body"
	mx={ 3 }
	pb={ 3 }
>
	Click me!
</Text>
```

Here, `component`, `variant`, `mx` and `pb` belong to Text component, and won't be propagated to the component instance.

Reply:
I'm deleting the spacing props in the line below:
```jsx
delete componentProps[ value ];
```

Reply:
Oh! Missed the `delete` part, sorry. 🚀

Reply:
yeah, it deserves a comment tbh :yesyes:

Reply:
https://github.com/Automattic/jetpack/pull/23682/commits/9f8b111cb0c488e40c33e15053a583e7d9cc7fd4